File: Single-Linked-List/delete_beg.py
# File Name: delete_beg.py
# Program to delete the first node of a singly linked list

class Node:
    def __init__(self, data):
        self.data = data
        self.next = None

head = None

# Function to delete beginning node
def delete_beg():
    global head
    if head is None:
        print("List is empty. Deletion not possible.")
        return
    head = head.next
    print("First node deleted successfully.")

if __name__ == "__main__":
    delete_beg()

File: Single-Linked-List/DeleteBeg.java
// File Name: DeleteBeg.java
// Program to delete the first node of a singly linked list

class Node {
    int data;
    Node next;

    Node(int data) {
        this.data = data;
        this.next = null;
    }
}

public class DeleteBeg {
    static Node head = null;

    // Function to delete beginning node
    static void delete_beg() {
        if (head == null) {
            System.out.println("List is empty. Deletion not possible.");
            return;
        }
        head = head.next;
        System.out.println("First node deleted successfully.");
    }

    public static void main(String[] args) {
        delete_beg();
    }
}
